hack/docker/postgres/10.2/scripts/replica/run.sh

```diff
@@ -1,69 +1,170 @@
 #!/usr/bin/env bash
 
-set -e
+set e
 
 echo "Running as Replica"
 
-mkdir -p "$PGDATA"
-rm -rf "$PGDATA"/*
-chmod 0700 "$PGDATA"
-
 # set password ENV
 export PGPASSWORD=${POSTGRES_PASSWORD:-postgres}
 
 export ARCHIVE=${ARCHIVE:-}
 
+gracefully_shutdown_host() {
+  echo "Gracefully shutting down database"
+
+  # start postgres server in background
+  postgres >/dev/null 2>&1 &
+
+  # Waiting for running Postgres
+  while true; do
+    echo "Attempting pg_isready on localhost"
+    pg_isready --timeout=2 &>/dev/null && break
+    sleep 2
+  done
+
+  # stop postgres server
+  pg_ctl -D "$PGDATA" stop >/dev/null 2>&1
+}
+
+take_pg_basebackup() {
+  mkdir -p "$PGDATA"
+  rm -rf "$PGDATA"/*
+  chmod 0700 "$PGDATA"
+
+  echo "Taking base backup."
+  pg_basebackup -X fetch --no-password --pgdata "$PGDATA" --username=postgres --host="$PRIMARY_HOST"
+}
+
+setup_postgresql_config() {
+  # setup recovery.conf
+  cp /scripts/replica/recovery.conf /tmp
+  echo "recovery_target_timeline = 'latest'" >>/tmp/recovery.conf
+  echo "archive_cleanup_command = 'pg_archivecleanup $PGWAL %r'" >>/tmp/recovery.conf
+
+  # primary_conninfo is used for streaming replication
+  echo "primary_conninfo = 'application_name=$HOSTNAME host=$PRIMARY_HOST'" >>/tmp/recovery.conf
+  mv /tmp/recovery.conf "$PGDATA/recovery.conf"
+
+  # setup postgresql.conf
+  cp /scripts/primary/postgresql.conf /tmp
+  echo "wal_level = replica" >>/tmp/postgresql.conf
+  echo "max_wal_senders = 99" >>/tmp/postgresql.conf
```

Comment:
```suggestion
  echo "max_wal_senders = 90" >>/tmp/postgresql.conf
```

Is a safe bet to keep it working with the max_conn setting of the primary (100) and admin connections (3).